### PR TITLE
[1LP][RFR] Automate test case: test_non_admin_user_reports_access_rest AND modify gen_data access_control functions

### PR DESCRIPTION
**Pull request by valaparthvi**
This PR brings the following changes:
1.  Create a separate file `intelligence/reports/test_reports.py` which includes all functional test cases of reports, this is a part of future refactoring of reports code.
2. Automate a test case: test_non_admin_user_reports_access_rest
3. Modify `users`, `roles`, `groups`, and `tenants` function to accept kwargs
4. Remove functions related to deprecated features.
5. Modify test cases using these functions to work well.

{{ pytest: cfme/tests/configure/test_rest_access_control.py cfme/tests/intelligence/reports/test_reports.py --use-template-cache -sqvvvv }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit c993894dfcf21d083a9dde935721e6809e945401:

`cfme/tests/intelligence/reports/test_reports.py`:
- :red_circle: Line 1:1: **F401** *'fauxfactory' imported but unused*
- :red_circle: Line 6:1: **F401** *'cfme.utils.appliance.implementations.ui.navigate_to' imported but unused*
- :red_circle: Line 8:1: **F401** *'cfme.utils.wait.wait_for_decorator' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/rest/gen_data.py [hunk 1]
```diff
@@ -368,25 +368,34 @@ def automation_requests_data(vm, requests_collection=False, approve=True, num=4)
...
+def groups(request, appliance, role, tenant, num=1, **kwargs):
     data = []
     for _ in range(num):
-        data.append({
-            "description": "group_description_{}".format(fauxfactory.gen_alphanumeric()),
-            "role": {"href": role.href},
-            "tenant": {"href": tenant.href}
-        })
+        data.append(
+            {
```
Comment (jaryn) [inline cfme/rest/gen_data.py]: This indentation style is rare to do in Python.

Comment (valaparthvi) [reply cfme/rest/gen_data.py]: I formatted the code with black and this is how it was formatted.

File: cfme/tests/intelligence/reports/test_reports.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        setup:
+            1. Create a user with privilege to access reports and REST.
+            2. Instantiate a MiqApi instance with the user.
+        testSteps:
+            1. Access all reports with the new user with the help of newly instantiated API.
+        expectedResults:
+            1. User should be able to access all reports.
+    """
+    report_data = rbac_api.collections.reports.all
+    assert_response(appliance)
```
Comment (izapolsk) [inline cfme/tests/intelligence/reports/test_reports.py]: QUESTION: @valaparthvi, what is that for ?

Comment (valaparthvi) [reply cfme/tests/intelligence/reports/test_reports.py]: That is to check if the last request sent via REST API received a successful 200 OK response.
Last request sent here is: `rbac_api.collections.reports.all`

**Pull request merged**